src/pages/index.js

```diff
@@ -52,47 +126,89 @@ function openProfilePopup() {
   editProfilePopup.open();
 }
 
-//Section Class
-
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: ({ name, link }) => {
-      const newCard = createCard({ name, link });
-      cardListSection.addItem(newCard);
-    },
-  },
-  cardList
-);
-
-cardListSection.renderItems();
-
-//PopupWithForm Class
+function handleProfileFormSubmit({ name, description }) {
+  editProfilePopup.setLoading(true);
+  api
+    .changeUserInfo(name, description)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editProfilePopup.setLoading(false, "Save");
+    });
+}
 
-const newCardPopup = new PopupWithForm(cardModalSelector, submitCard);
-const previewImagePopup = new PopupWithImage(previewImageModal);
+//Profile Set Event Listeners
+profileEditButton.addEventListener("click", openProfilePopup);
+profileImageButton.addEventListener("click", openChangeProfilePicturePopup);
 
-addNewCardButton.addEventListener("click", () => {
-  addCardFormValidator.toggleButtonState();
-  newCardPopup.open();
-});
+function openChangeProfilePicturePopup() {
+  changeProfilePictureFormValidator.toggleButtonState();
+  changeProfilePicturePopup.open();
+}
 
 //Card Functions
+function submitCard({ title, url }) {
+  newCardPopup.setLoading(true);
+  api
+    .addCard(title, url)
+    .then((data) => {
+      const newCard = createCard(data, userId);
+      cardListSection.addItem(newCard);
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      newCardPopup.setLoading(false, "Create");
+    });
+}
 
-function createCard({ name, link }) {
+function createCard(cardData, userId) {
   const cardElement = new Card(
-    { name, link },
-    "#card-template",
-    ({ name, link }) => {
-      previewImagePopup.open({ name, link });
-    }
+    {
+      cardData: cardData,
+      cardSelector: "#card-template",
+      handlePreviewImage: ({ name, link }) => {
+        previewImagePopup.open({ name, link });
+      },
+      handleLikeButton: (cardId, isLiked) => {
+        api.changeLikeNumber(cardId, isLiked).then((data) => {
+          cardElement.setLikes(data.likes);
+        });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**

Reply:
Updated.